plan/range.go

```diff
@@ -116,6 +117,94 @@ func (r *rangeBuilder) build(expr ast.ExprNode) []rangePoint {
 	return fullRange
 }
 
+func (r *rangeBuilder) newBuild(expr expression.Expression) []rangePoint {
+	switch x := expr.(type) {
+	case *expression.Column:
+		return r.buildFromColumn(x)
+	case *expression.ScalarFunction:
```

Comment:
If the constant value is 0 or null, we should return empty range, otherwise return full range.